# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1234849393 18000
#      Tue Feb 17 00:43:13 2009 -0500
# Node ID aeedc045cfe316ab3e67ae8977a6a6eb13438f10
# Parent  d3382daaf4d2bee87c0654b5297b5ae13a96f875
toplev.cc (Foctave_config_info): add CARBON_LIBS, X11_INCFLAGS, and X11_LIBS to the struct

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,13 @@
+2009-02-17  John W. Eaton  <jwe@octave.org>
+
+	* Makeconf.in (do-subst-config-vals): Substitute CARBON_LIBS,
+	X11_INCFLAGS, and X11_LIBS.
+
 2009-02-17  Thomas Treichl  <Thomas.Treichl@gmx.net>
 
 	* Makeconf.in: Substitute CARBON_LIBS.
 	* configure.in: Use CARBON_LIBS instead of LIBS for framework Carbon.
 
 	* configure.in: Fix test for X11 if "--without-x" is given.
 
 2009-02-16  Jaroslav Hajek  <highegg@gmail.com>
diff --git a/Makeconf.in b/Makeconf.in
--- a/Makeconf.in
+++ b/Makeconf.in
@@ -471,16 +471,17 @@ echo "making $@ from $<"
   -e "s|%OCTAVE_CONF_ALL_CFLAGS%|\"${ALL_CFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_ALL_CXXFLAGS%|\"${ALL_CXXFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_ALL_FFLAGS%|\"${ALL_FFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_ALL_LDFLAGS%|\"${ALL_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_AR%|\"${AR}\"|" \
   -e "s|%OCTAVE_CONF_ARFLAGS%|\"${ARFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_BLAS_LIBS%|\"${BLAS_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_CANONICAL_HOST_TYPE%|\"${canonical_host_type}\"|" \
+  -e "s|%OCTAVE_CONF_CARBON_LIBS%|\"${CARBON_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_CC%|\"${CC}\"|" \
   -e "s|%OCTAVE_CONF_CC_VERSION%|\"${CC_VERSION}\"|" \
   -e "s|%OCTAVE_CONF_CFLAGS%|\"${CFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_CPICFLAG%|\"${CPICFLAG}\"|" \
   -e "s|%OCTAVE_CONF_CPPFLAGS%|\"${CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_CURL_LIBS%|\"${CURL_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_CXX%|\"${CXX}\"|" \
   -e "s|%OCTAVE_CONF_CXXCPP%|\"${CXXCPP}\"|" \
@@ -539,16 +540,18 @@ echo "making $@ from $<"
   -e "s|%OCTAVE_CONF_SH_LD%|\"${SH_LD}\"|" \
   -e "s|%OCTAVE_CONF_SH_LDFLAGS%|\"${SH_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_SONAME_FLAGS%|\"${SONAME_FLAGS}\"|" \
   -e "s|%OCTAVE_CONF_STATIC_LIBS%|\"${STATIC_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_UGLY_DEFS%|\"${UGLY_DEFS}\"|" \
   -e "s|%OCTAVE_CONF_USE_64_BIT_IDX_T%|\"${USE_64_BIT_IDX_T}\"|" \
   -e "s|%OCTAVE_CONF_VERSION%|\"${version}\"|" \
   -e "s|%OCTAVE_CONF_ENABLE_DYNAMIC_LINKING%|\"${ENABLE_DYNAMIC_LINKING}\"|" \
+  -e "s|%OCTAVE_CONF_X11_INCFLAGS%|\"${X11_INCFLAGS}\"|" \
+  -e "s|%OCTAVE_CONF_X11_LIBS%|\"${X11_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_XTRA_CFLAGS%|\"${XTRA_CFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_XTRA_CXXFLAGS%|\"${XTRA_CXXFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_YACC%|\"${YACC}\"|" \
   -e "s|%OCTAVE_CONF_YFLAGS%|\"${YFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_config_opts%|\"${config_opts}\"|" | \
   $(SED)  -e "s|%OCTAVE_CONF_DEFS%|\"${UGLY_DEFS}\"|" > $@-t
 $(simple-move-if-change-rule)
 endef
diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,15 @@
+2009-02-17  John W. Eaton  <jwe@octave.org>
+
+	* toplev.cc (octave_config_info): Add CARBON_LIBS, X11_INCFLAGS,
+	and X11_LIBS to the struct.
+	* oct-conf.h.in (OCTAVE_CONF_CARBON_LIBS,
+	OCTAVE_CONF_X11_INCFLAGS, OCTAVE_CONF_X11_LIBS): New macros.
+
 2009-02-17  Thomas Treichl  <Thomas.Treichl@gmx.net>
 
 	* Makefile.in (octave$(EXEEXT)): Link with CARBON_LIBS.
 
 2009-02-16  John W. Eaton  <jwe@octave.org>
 
 	* lex.l (next_token_can_follow_bin_op):
 	Push all characters read on to buffer stack.
diff --git a/src/oct-conf.h.in b/src/oct-conf.h.in
--- a/src/oct-conf.h.in
+++ b/src/oct-conf.h.in
@@ -48,16 +48,20 @@ along with Octave; see the file COPYING.
 #ifndef OCTAVE_CONF_AR
 #define OCTAVE_CONF_AR %OCTAVE_CONF_AR%
 #endif
 
 #ifndef OCTAVE_CONF_BLAS_LIBS
 #define OCTAVE_CONF_BLAS_LIBS %OCTAVE_CONF_BLAS_LIBS%
 #endif
 
+#ifndef OCTAVE_CONF_CARBON_LIBS
+#define OCTAVE_CONF_CARBON_LIBS %OCTAVE_CONF_CARBON_LIBS%
+#endif
+
 #ifndef OCTAVE_CONF_CC
 #define OCTAVE_CONF_CC %OCTAVE_CONF_CC%
 #endif
 
 #ifndef OCTAVE_CONF_CC_VERSION
 #define OCTAVE_CONF_CC_VERSION %OCTAVE_CONF_CC_VERSION%
 #endif
 
@@ -304,16 +308,24 @@ along with Octave; see the file COPYING.
 #ifndef OCTAVE_CONF_USE_64_BIT_IDX_T
 #define OCTAVE_CONF_USE_64_BIT_IDX_T %OCTAVE_CONF_USE_64_BIT_IDX_T%
 #endif
 
 #ifndef OCTAVE_CONF_ENABLE_DYNAMIC_LINKING
 #define OCTAVE_CONF_ENABLE_DYNAMIC_LINKING %OCTAVE_CONF_ENABLE_DYNAMIC_LINKING%
 #endif
 
+#ifndef OCTAVE_CONF_X11_INCFLAGS
+#define OCTAVE_CONF_X11_INCFLAGS %OCTAVE_CONF_X11_INCFLAGS%
+#endif
+
+#ifndef OCTAVE_CONF_X11_LIBS
+#define OCTAVE_CONF_X11_LIBS %OCTAVE_CONF_X11_LIBS%
+#endif
+
 #ifndef OCTAVE_CONF_XTRA_CFLAGS
 #define OCTAVE_CONF_XTRA_CFLAGS %OCTAVE_CONF_XTRA_CFLAGS%
 #endif
 
 #ifndef OCTAVE_CONF_XTRA_CXXFLAGS
 #define OCTAVE_CONF_XTRA_CXXFLAGS %OCTAVE_CONF_XTRA_CXXFLAGS%
 #endif
 
diff --git a/src/toplev.cc b/src/toplev.cc
--- a/src/toplev.cc
+++ b/src/toplev.cc
@@ -1143,16 +1143,17 @@ specified option.\n\
     {
       { false, "ALL_CFLAGS", OCTAVE_CONF_ALL_CFLAGS },
       { false, "ALL_CXXFLAGS", OCTAVE_CONF_ALL_CXXFLAGS },
       { false, "ALL_FFLAGS", OCTAVE_CONF_ALL_FFLAGS },
       { false, "ALL_LDFLAGS", OCTAVE_CONF_ALL_LDFLAGS },
       { false, "AR", OCTAVE_CONF_AR },
       { false, "ARFLAGS", OCTAVE_CONF_ARFLAGS },
       { false, "BLAS_LIBS", OCTAVE_CONF_BLAS_LIBS },
+      { false, "CARBON_LIBS", OCTAVE_CONF_CARBON_LIBS },
       { false, "CC", OCTAVE_CONF_CC },
       { false, "CC_VERSION", OCTAVE_CONF_CC_VERSION },
       { false, "CFLAGS", OCTAVE_CONF_CFLAGS },
       { false, "CPICFLAG", OCTAVE_CONF_CPICFLAG },
       { false, "CPPFLAGS", OCTAVE_CONF_CPPFLAGS },
       { false, "CURL_LIBS", OCTAVE_CONF_CURL_LIBS },
       { false, "CXX", OCTAVE_CONF_CXX },
       { false, "CXXCPP", OCTAVE_CONF_CXXCPP },
@@ -1202,16 +1203,18 @@ specified option.\n\
       { false, "SHLEXT", OCTAVE_CONF_SHLEXT },
       { false, "SHLEXT_VER", OCTAVE_CONF_SHLEXT_VER },
       { false, "SH_LD", OCTAVE_CONF_SH_LD },
       { false, "SH_LDFLAGS", OCTAVE_CONF_SH_LDFLAGS },
       { false, "SONAME_FLAGS", OCTAVE_CONF_SONAME_FLAGS },
       { false, "STATIC_LIBS", OCTAVE_CONF_STATIC_LIBS },
       { false, "UGLY_DEFS", OCTAVE_CONF_UGLY_DEFS },
       { false, "USE_64_BIT_IDX_T", OCTAVE_CONF_USE_64_BIT_IDX_T },
+      { false, "X11_INCFLAGS", OCTAVE_CONF_X11_INCFLAGS },
+      { false, "X11_LIBS", OCTAVE_CONF_X11_LIBS },
       { false, "XTRA_CFLAGS", OCTAVE_CONF_XTRA_CFLAGS },
       { false, "XTRA_CXXFLAGS", OCTAVE_CONF_XTRA_CXXFLAGS },
       { false, "YACC", OCTAVE_CONF_YACC },
       { false, "YFLAGS", OCTAVE_CONF_YFLAGS },
       { false, "api_version", OCTAVE_API_VERSION },
       { true, "archlibdir", OCTAVE_ARCHLIBDIR },
       { true, "bindir", OCTAVE_BINDIR },
       { false, "canonical_host_type", OCTAVE_CANONICAL_HOST_TYPE },
